src/clj/puppetlabs/puppetserver/certificate_authority.clj

```diff
@@ -674,26 +682,25 @@
                        :msg (i18n/tru "Found extensions that are not permitted: {0}"
                                  (str/join ", " bad-extension-oids))})))))
 
-(schema/defn validate-dns-alt-names!
+(schema/defn validate-subject-alt-names!
   "Validate that the provided Subject Alternative Names extension is valid for
   a cert signed by this CA. This entails:
-    * Only DNS alternative names are allowed, no other types
+    * Only DNS and IP alternative names are allowed, no other types
     * Each DNS name does not contain a wildcard character (*)"
   [{value :value} :- Extension]
   (let [name-types (keys value)
-        names      (:dns-name value)]
-    (when-not (and (= (count name-types) 1)
-                   (= (first name-types) :dns-name))
+        dns-names (:dns-name value)
+        ip-names (:ip value)]
+    (when-not (every? #{:dns-name :ip} name-types)
       (sling/throw+
         {:kind :invalid-alt-name
-         :msg (i18n/tru "Only DNS names are allowed in the Subject Alternative Names extension")}))
-
-    (doseq [name names]
-      (when (.contains name "*")
+         :msg (i18n/tru "Only DNS and IP names are allowed in the Subject Alternative Names extension")}))
+    (doseq [dns-name dns-names]
+      (when (.contains dns-name "*")
```

Comment:
`*` is disallowed for only dns-names?

Reply:
As far as I know. IP addresses can't contain "*" and be valid anyway...i think.